src/librustc/hir/check_attr.rs

```diff
@@ -93,70 +93,102 @@ struct CheckAttrVisitor<'tcx> {
 impl CheckAttrVisitor<'tcx> {
     /// Checks any attribute.
     fn check_attributes(&self, item: &hir::Item, target: Target) {
-        if target == Target::Fn || target == Target::Const {
-            self.tcx.codegen_fn_attrs(self.tcx.hir().local_def_id(item.hir_id));
-        } else if let Some(a) = item.attrs.iter().find(|a| a.check_name(sym::target_feature)) {
-            self.tcx.sess.struct_span_err(a.span, "attribute should be applied to a function")
-                .span_label(item.span, "not a function")
-                .emit();
-        }
-
+        let mut is_valid = true;
         for attr in &item.attrs {
```

Comment:
While we're at it... Maybe reformulate this with `item.attrs.all(...)`?

Reply:
I personally prefer the explicit loop here.

Reply:
I've not changed this because `.all(...)` stops as soon as the first attribute returns false, so we don't report multiple errors if I make this change.